xonsh/xoreutils/cat.py

```diff
@@ -49,15 +52,11 @@ def _cat_single_file(opts, fname, stdin, out, err, line_count=1):
         print("cat: No such file or directory: {}".format(fname), file=err)
         return True, line_count
     else:
-        fstat = os.stat(fname)
-        file_size = fstat.st_size
-        if file_size == 0 and not stat.S_ISREG(fstat.st_mode):
-            file_size = None
         fobj = open(fname, "rb")
         f = xproc.NonBlockingFDReader(fobj.fileno(), timeout=0.1)
     sep = os.linesep.encode(enc, enc_errors)
     last_was_blank = False
-    while file_size is None or read_size < file_size:
+    while True:
```

Comment:
I think some of these conditions might need to be implemented in the body for safety

Reply:
pinginf this again @virusbb001

Reply:
Thanks for reviewing and sorry for the delayed response.
I think you are right but I don't know what is the good condition for safety...
Does checking `endnow` meet safety?

Reply:
Is there a reason the original check doesn't apply anymore is what I was asking. I think it could be restored.

Reply:
The original check ( 344671ba ) caused #3182 because `stat.S_ISREG(os.stat("/proc/cpuinfo").st_mode)` returns 0 and the file size of `/proc/cpuinfo` is 0.
But the previous of original check( 7b7a9a30 ) works fine, so I think it can be restored. I will try it.